boa/src/builtins/nan/mod.rs

```diff
@@ -0,0 +1,8 @@
+use crate::{builtins::value::Value, BoaProfiler};
+
+/// Initialise the `NaN` property on the global object.
+#[inline]
+pub fn init(global: &Value) {
+    let _timer = BoaProfiler::global().start_event("NaN", "init");
+    global.set_field("NaN", Value::from(f64::NAN));
+}
```

Comment:
If you ask me I would remove all these `init` functions as all of them are one-liners binding the objects on the provided global object.

I would just put in here "boa/src/builtins/mod.rs" this:
```rust
pub fn init(global: &Value) {
    global.set_field("Function", Function::new());
    global.set_field("BigInt", BigInt::new());
    ...
    // instead of
    //BigInt::init(global);
    //Function::init(global);
    // ...
}```

Reply:
> If you ask me I would remove all these `init` functions as all of them are one-liners binding the objects on the provided global object.
> 
> I would just put in here "boa/src/builtins/mod.rs", something like this:
> ```rust
> pub fn init(global: &Value) {
>     global.set_field("Function", Function::new());
>     global.set_field("BigInt", BigInt::new());
>     ...
>     // instead of
>     //BigInt::init(global);
>     //Function::init(global);
>     // ...
> }```


Yes, I think this is a good idea. We should have an `InitBuiltin` trait, that could be implemented by external libraries too, that just have an `init()` function and return a list of `<&str, Value>` that we would then add to the global object, maybe in parallel (or at least they could be generated in parallel).

Reply:
BTW, something similar was commented [here](https://github.com/boa-dev/boa/pull/419#discussion_r438685148) too.